packages/ui/__tests__/Card.test.tsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react'
+import { mountAndCheckA11Y } from '@hazelcast/test-helpers'
+
+import { Card, CardProps, Icon, IconProps } from '../src'
+import { Database } from 'react-feather'
+
+describe('Card', () => {
+  const cardTitle = 'Card title'
+  const cardContent = 'Card content'
+
+  const cardLayoutTestData: CardProps[] = [
+    {
+      title: cardTitle,
+    },
+    {
+      title: cardTitle,
+      icon: Database,
+    },
+    {
+      title: cardTitle,
+      icon: Database,
+      separator: true,
+    },
+    {
+      title: cardTitle,
+      icon: Database,
+      separator: true,
+    },
+  ]
+
+  it.each(cardLayoutTestData)('Renders correct layout for specific component configurations.', async (props) => {
+    const wrapper = await mountAndCheckA11Y(<Card {...props}>{cardContent}</Card>)
+
+    const icon = wrapper.find(Icon)
+    const title = wrapper.findDataTest('card-heading')
+    const separator = wrapper.findDataTest('card-separator')
+
+    if (props.title) {
+      expect(title.text()).toEqual(props.title)
+      expect(title.getDOMNode().tagName).toEqual('H3')
+    } else {
+      expect(title.length).toEqual(0)
```

Comment:
nit: Personally, I find using `exists()` - `expect(title.exists()).toBe(false)` - to be a bit more readable :)

Reply:
I just wanted to be consistent, but maybe readability is better, you're right :) I replaced all "0 length" checks with `exists()`, thanks!